runtime/common/src/migrations/liquidity_pools_v2.rs

```diff
@@ -0,0 +1,846 @@
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_types::domain_address::Domain;
+#[cfg(feature = "try-runtime")]
+use frame_support::pallet_prelude::{Decode, Encode};
+use frame_support::{
+	pallet_prelude::ValueQuery,
+	storage_alias,
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+	Blake2_128Concat,
+};
+use pallet_axelar_router::AxelarId;
+use sp_runtime::DispatchError;
+use sp_std::marker::PhantomData;
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::routing::RouterId;
+
+const LOG_PREFIX: &str = "LpV2";
+
+fn maybe_router_id(domain: Domain) -> Option<RouterId> {
+	match domain.get_evm_chain_id() {
+		None => {
+			log::error!(
+				"{LOG_PREFIX}: Failed derive chain_id from domain {domain:?} because it's not EVM"
+			);
+			None
+		}
+		Some(id) => Some(RouterId::Axelar(AxelarId::Evm(id))),
+	}
+}
+
+pub mod kill_relayer_list {
+	use frame_support::traits::OnRuntimeUpgrade;
+	#[cfg(feature = "try-runtime")]
+	use frame_support::{dispatch::DispatchResult, storage::with_storage_layer};
+	#[cfg(feature = "try-runtime")]
+	use sp_arithmetic::traits::Zero;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec;
+
+	use super::{types::v0, *};
+	use crate::migrations::nuke::storage_clean_res_log;
+
+	const LOG_PREFIX: &str = "ClearRelayerList";
+
+	pub struct Migration<T>(PhantomData<T>);
+
+	impl<T> OnRuntimeUpgrade for Migration<T>
+	where
+		T: pallet_liquidity_pools_gateway::Config,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			let res = v0::RelayerList::<T>::clear(20, None);
+			storage_clean_res_log(&res, "RelayerList", LOG_PREFIX);
+
+			log::info!("{LOG_PREFIX}: Migration done!");
+
+			T::DbWeight::get().reads_writes(res.loops.into(), res.unique.into())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			let mut cleared: bool = false;
+
+			// Need to rollback in order to be NOOP
+			let _ = with_storage_layer(|| -> DispatchResult {
+				cleared = v0::RelayerList::<T>::clear(20, None).maybe_cursor.is_none();
+				Err(DispatchError::Other("Reverting on purpose"))
+			});
+			assert!(cleared);
+
+			log::info!("{LOG_PREFIX}: Pre checks done!");
+
+			Ok(vec![])
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(_: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+			assert!(v0::RelayerList::<T>::iter_keys().count().is_zero());
+			log::info!("{LOG_PREFIX}: Post checks done!");
+
+			Ok(())
+		}
+	}
+}
+
+pub mod v2_update_message_queue {
```

Comment:
Note: Dev and Demo were already upgraded to the intermediary storage version 2 which still supported `gateway::DomainRouters` storage but removed all routers but the `AxelarEvm` ones. The corresponding commit hash is https://github.com/centrifuge/centrifuge-chain/pull/1966/commits/8ec41e92a8b0bd31fef1c90e6a47cbcd48e771e6

Therefore, we need to provide migrations from v0 and v2.